plugins/resteasy-http/src/main/java/org/radargun/service/RESTEasyHTTPService.java

```diff
@@ -0,0 +1,198 @@
+package org.radargun.service;
+
+import java.lang.reflect.Type;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jboss.resteasy.client.jaxrs.BasicAuthentication;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder.HostnameVerificationPolicy;
+import org.jboss.resteasy.client.jaxrs.engines.URLConnectionEngine;
+import org.radargun.Service;
+import org.radargun.config.Converter;
+import org.radargun.config.Property;
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.traits.Lifecycle;
+import org.radargun.traits.ProvidesTrait;
+import org.radargun.utils.TimeConverter;
+
+/**
+ * REST client service using the JAX-RS 2.0 client api with the RESTEasy engine based on
+ * java.net.HttpURLConnection
+ * 
+ * @author Alan Field &lt;afield@redhat.com&gt;
+ */
+@Service(doc = "ApacheHTTP REST client")
+public class RESTEasyHTTPService implements Lifecycle {
+   private static final Log log = LogFactory.getLog(RESTEasyHTTPService.class);
+   private static final String PATTERN_STRING = "(\\[([0-9A-Fa-f:]+)\\]|([^:/?#]*))(?::(\\d*))?";
+   private static final Pattern ADDRESS_PATTERN = Pattern.compile(PATTERN_STRING);
+   private static final int DEFAULT_PORT = 8080;
+   private ResteasyClient httpClient = null;
+
+   @Property(doc = ". Defaults to 'rest'.")
+   private String serverLocation = "rest";
```

Comment:
Could we use serverPath, or rootPath or something like that? And please add more descriptive doc.